#16273: Fix bug in method getUrlInStore() of product model

vasilii-b opened the pull request:
### Description
This PR reference changes how/what is returned when `product` model method `getUrlInStore()` is called `$product->getUrlInStore()`.
Problem is in how the current store is checked and what is returned based on it.

### Fixed Issues
1. magento/magento2#16273
2. Having set _Add store code to URLs_ in Admin Configuration to _NO_, store code is still shown in the product URL.

### Manual testing scenarios

__For issue 1__ - please see the issue description.
__For issue 2__:

1. Go to Admin -> Stores -> Configuration -> Catalog -> Web -> Url Options -> Add Store Code to Urls and set it to NO.
2. In some `.phtml` file, list.phtml for example, call method `getUrlInStore()` for a product.

#### Expected Result
Store code is not added into product url
```
https://www.domain.com/my-product.html
```

#### Actual Result
Store code is added into product url
```
https://www.domain.com/my-product.html?___store=default
```

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

victor-v-rad committed:
MAGETWO-74021: "Catalog Products List" widget does not displays on frontend

le0n4eg committed:
MAGETWO-87589: [Magento Cloud] - Issue with polluted database when updating product attributes through API

le0n4eg committed:
MAGETWO-87589: [Magento Cloud] - Issue with polluted database when updating product attributes through API

Commit:
#15205: Postpone instantiation of session config by using a proxy

RuslanKostiv1 committed:
Merge remote-tracking branch 'mainline/2.2-develop' into MAGETWO-89238-2.2

magento-engcom-team committed:
ENGCOM-1703: Move breadcrumb json configuration to viewmodel #15521

 - Merge Pull Request magento/magento2#15521 from diedburn/magento2:fix_breadcrumbs_encoding
 - Merged commits:
   1. 213e11d576d72a3d6bd2efd1a2f309f91d200d5d
   2. 603de52d736e5a7e3e6473661823749ac3159961
   3. e95f3f7a1d790d661a024e12ad6bf34d33bb9485

victor-v-rad committed:
MAGETWO-74021: "Catalog Products List" widget does not displays on frontend

victor-v-rad committed:
MAGETWO-74021: "Catalog Products List" widget does not displays on frontend

victor-v-rad committed:
Merge remote-tracking branch 'mainline/2.2-develop' into MAGETWO-74021

RuslanKostiv1 committed:
MAGETWO-89238: [Sales email] Disabled invoice emails produce performance issue

RuslanKostiv1 committed:
MAGETWO-89238: [Sales email] Disabled invoice emails produce performance issue

Commit:
#15929: Apply changes to HTTP response also to its subclasses

Commit:
#15929: Reset changes to HTTP response classes

RuslanKostiv1 committed:
MAGETWO-89238: [Sales email] Disabled invoice emails produce performance issue

RuslanKostiv1 committed:
MAGETWO-89238: [Sales email] Disabled invoice emails produce performance issue

RuslanKostiv1 committed:
MAGETWO-89238: [Sales email] Disabled invoice emails produce performance issue

RuslanKostiv1 committed:
MAGETWO-89238: [Sales email] Disabled invoice emails produce performance issue

RuslanKostiv1 committed:
MAGETWO-89238: [Sales email] Disabled invoice emails produce performance issue

RuslanKostiv1 committed:
MAGETWO-89238: [Sales email] Disabled invoice emails produce performance issue

denteblu committed:
during product import, use correct error message for duplicate error key

svitja committed:
MAGETWO-92504: Products sorting in category is reverted to sort by product ID

ishakhsuvarov committed:
Updated according to coding style requirements

RuslanKostiv1 committed:
MAGETWO-89238: [Sales email] Disabled invoice emails produce performance issue

RuslanKostiv1 committed:
MAGETWO-89238: [Sales email] Disabled invoice emails produce performance issue

DianaRusin committed:
MAGETWO-88102: [Magento Cloud] Catalog_category_product_index table doesn't update after deleting category

DianaRusin committed:
MAGETWO-88102: [Magento Cloud] Catalog_category_product_index table doesn't update after deleting category

RuslanKostiv1 committed:
MAGETWO-89238: [Sales email] Disabled invoice emails produce performance issue

RuslanKostiv1 committed:
MAGETWO-89238: [Sales email] Disabled invoice emails produce performance issue

RuslanKostiv1 committed:
MAGETWO-89238: [Sales email] Disabled invoice emails produce performance issue

RuslanKostiv1 committed:
MAGETWO-89238: [Sales email] Disabled invoice emails produce performance issue

zakdma committed:
MAGETWO-87721: Custom Options are corruputed when saving product to a different website

magento-engcom-team committed:
ENGCOM-1911: FIX for issue #15501 - M2.2.4 missing meta title tag and doesn't show… #15532

 - Merge Pull Request magento/magento2#15532 from phoenix128/magento2:issue-15501-missing-meta-title-tag
 - Merged commits:
   1. 0a38b845508577e38d79280288a4fc716f2d1108
   2. 85ac3be16708f9b7084481e522d29320e4a315e3
   3. 432cf007f79937437620f18ace00010b1a61a4ef
   4. 1413d65946c2e3e076a0605d2e5c739e0b339d83

RuslanKostiv1 committed:
MAGETWO-89238: [Sales email] Disabled invoice emails produce performance issue

RuslanKostiv1 committed:
MAGETWO-89238: [Sales email] Disabled invoice emails produce performance issue

svitja committed:
MAGETWO-75427: Hint on product option with Maximum Characters should count down as more characters are entered

AVoskoboinikov committed:
MAGETWO-92693: Some improvements on product create|edit page in admin area

RuslanKostiv1 committed:
MAGETWO-89238: [Sales email] Disabled invoice emails produce performance issue

AVoskoboinikov committed:
MAGETWO-92693: Some improvements on product create|edit page in admin area

RuslanKostiv1 committed:
MAGETWO-89238: [Sales email] Disabled invoice emails produce performance issue

AVoskoboinikov committed:
MAGETWO-92693: Some improvements on product create|edit page in admin area

AVoskoboinikov committed:
MAGETWO-92693: Some improvements on product create|edit page in admin area

zakdma committed:
MAGETWO-87721: Custom Options are corruputed when saving product to a different website

RuslanKostiv1 committed:
Merge remote-tracking branch 'mainline/2.2-develop' into MAGETWO-89238-2.2

rsantellan committed:
When searching for the title if search for all the segments that has a discount title making it possible displaying custom discounts.

YevSent committed:
MAGETWO-86658: Admin global search preview works inconsistently.

 - Added fulltext search by title in the CMS pages grid

DianaRusin committed:
MAGETWO-88102: [Magento Cloud] Catalog_category_product_index table doesn't update after deleting category

StasKozar committed:
Merge remote-tracking branch 'origin/MAGETWO-75427' into 2.2-develop-pr31

StasKozar committed:
Merge remote-tracking branch 'origin/MAGETWO-92504' into 2.2-develop-pr31

StasKozar committed:
Merge remote-tracking branch 'origin/MAGETWO-88102' into 2.2-develop-pr31

lfluvisotto committed:
Fix case mismatch call (class/method)

magento-engcom-team committed:
ENGCOM-2007: [Backport] Fix issue #14895 - Change Password warning message appear two times #15774

 - Merge Pull Request magento/magento2#15774 from sanjay-wagento/magento2:2.2-develop-PR-port-14897
 - Merged commits:
   1. b2cc966da3a6bcc925035e12a950e636847326f8
   2. cfe58e16b5159f97fbf8b1246a6e5d75caaaf9e0

zakdma committed:
MAGETWO-92751: Shipping page contains the equal shipping addresses after adding a user.

AVoskoboinikov committed:
MAGETWO-92693: Some improvements on product create|edit page in admin area

StasKozar committed:
MAGETWO-73967: [2.2.x] - [Github]Can not save attribute #5907

AVoskoboinikov committed:
MAGETWO-92693: Some improvements on product create|edit page in admin area

AVoskoboinikov committed:
MAGETWO-92693: Some improvements on product create|edit page in admin area

AVoskoboinikov committed:
MAGETWO-92693: Some improvements on product create|edit page in admin area

OlgaVasyltsun committed:
MAGETWO-90837: Can't save customer address on adminhtml when country restriction is enabled

OlgaVasyltsun committed:
Merge remote-tracking branch 'origin/2.2-develop' into MAGETWO-90837

svitja committed:
Merge remote-tracking branch 'origin/2.2-develop' into MAGETWO-92504

# Conflicts:
#	dev/tests/acceptance/tests/functional/Magento/FunctionalTest/Catalog/Data/ProductData.xml

svitja committed:
Merge remote-tracking branch 'origin/MAGETWO-92504' into 2.2-develop-pr31

viktym committed:
MAGETWO-92820: Once Payflow Pro payment is Declined, customer cannot continue checkout

DianaRusin committed:
Merge remote-tracking branch 'origin/MAGETWO-73967' into 2.2-develop-pr31

YevSent committed:
MAGETWO-92280: [B2B] Unable to add to cart grouped product by SKU with enabled Shared Catalog

 - Added quote ID to request params

StasKozar committed:
MAGETWO-92312: Wrong price and Cart Subtotal at mini shopping cart

svitja committed:
MAGETWO-81926: Category hierarchy isn't updated on Product Admin page

svitja committed:
MAGETWO-83992: Unable to delete Fixed Tax Attribute from attribute set

svitja committed:
MAGETWO-81926: Category hierarchy isn't updated on Product Admin page

svitja committed:
MAGETWO-83992: Unable to delete Fixed Tax Attribute from attribute set

zakdma committed:
Merge remote-tracking branch 'origin/2.2-develop' into MAGETWO-92751

vrann committed:
Merge remote-tracking branch 'bulk/bulk-api-backport' into bulk-api-backported

svitja committed:
MAGETWO-83992: Unable to delete Fixed Tax Attribute from attribute set

Commit:
PHPDoc

magento-engcom-team committed:
ENGCOM-2034: PHPDoc #16215

 - Merge Pull Request magento/magento2#16215 from lfluvisotto/magento2:2.2-develop-phpdoc
 - Merged commits:
   1. 0e9ed86353874d6fa8c9c5bbec2612bfeba5583c

svitja committed:
MAGETWO-83992: Unable to delete Fixed Tax Attribute from attribute set

vrann committed:
magento-engcom/bulk-api#20 Backport Change Webapi extension to use Inject Pool logic

- remove backward incompatible changes

vrann committed:
Merge remote-tracking branch 'bulk/bulk-api-backport' into bulk-api-backported

vrann committed:
magento-engcom/bulk-api#20 Backport Change Webapi extension to use Inject Pool logic

- add needed dependencies

vrann committed:
magento-engcom/bulk-api#20 Backport Change Webapi extension to use Inject Pool logic

- add needed dependencies

vrann committed:
Merge remote-tracking branch 'bulk/bulk-api-backport' into bulk-api-backported

AnshuMishra17 committed:
Admin controller product set save refactor

vrann committed:
magento-engcom/bulk-api#20 Backport Change Webapi extension to use Inject Pool logic

- add needed dependencies

vrann committed:
Merge remote-tracking branch 'bulk/bulk-api-backport' into bulk-api-backported

vrann committed:
magento-engcom/bulk-api#20 Backport Change Webapi extension to use Inject Pool logic

- add needed dependencies

vrann committed:
magento-engcom/bulk-api#20 Backport Change Webapi extension to use Inject Pool logic

- fix formatting

vrann committed:
Merge remote-tracking branch 'bulk/bulk-api-backport' into bulk-api-backported

vrann committed:
magento-engcom/bulk-api#20 Backport Change Webapi extension to use Inject Pool logic

- update version

vrann committed:
Merge remote-tracking branch 'bulk/bulk-api-backport' into bulk-api-backported

svitja committed:
MAGETWO-81926: Category hierarchy isn't updated on Product Admin page

zakdma committed:
MAGETWO-92751: Shipping page contains the equal shipping addresses after adding a user.

zakdma committed:
Merge remote-tracking branch 'origin/2.2-develop' into MAGETWO-92751

Commit:
Fixed according to Backward compatible development guide

magento-engcom-team committed:
ENGCOM-1892: Create ability to set is_visible_on_front to order status history comment #15637

 - Merge Pull Request magento/magento2#15637 from markoshust/magento2:feature/order-history-comment-is-visible-on-front
 - Merged commits:
   1. 8aea8ca5c5250c15e3b2379c66ac492f44737c69
   2. 594203c52e82ac26fa86c09d5f3f9ab36ec20492

svitja committed:
MAGETWO-81926: Category hierarchy isn't updated on Product Admin page

DianaRusin committed:
Merge remote-tracking branch 'origin/MAGETWO-92312' into 2.2-develop-pr31

DianaRusin committed:
Merge remote-tracking branch 'origin/MAGETWO-90837' into 2.2-develop-pr31

OlgaVasyltsun committed:
MAGETWO-91163: Image doesn't load on frontend PDP if the product name contains double quote "

zakdma committed:
Merge remote-tracking branch 'origin/2.2-develop' into 2.2-develop-pr31

magento-engcom-team committed:
ENGCOM-2042: Fix #15627: Product order in category changed in Magento 2.2.4 #15629

 - Merge Pull Request magento/magento2#15629 from dverkade/magento2:2.2-Fix-15627-Product-order-in-category-changed
 - Merged commits:
   1. 8981225996e92af97671c67727be35af42591e94

sanganinamrata committed:
Removed double occurrence of 'it' from sentences.

sanganinamrata committed:
Correct grammer mistakes in sentences.

sanganinamrata committed:
Merge branch '2.2-develop' into nps-sentence-fix-2

sanganinamrata committed:
Fixed wrong input string into csv of Magento_Paypal module for en_US locale.

magento-engcom-team committed:
ENGCOM-2045: Fix case mismatch call (class/method) #16141

 - Merge Pull Request magento/magento2#16141 from lfluvisotto/magento2:2.2-develop-case-mismatch-call
 - Merged commits:
   1. a09ee80a89dc97d59f5799038ebd4ff6d9b100fc

gelanivishal committed:
Fixed typo error

IvanPletnyov committed:
ENGCOM-1703: Move breadcrumb json configuration to viewmodel (fix badly json)

magento-engcom-team committed:
ENGCOM-2055: Fixed typo error #16247

 - Merge Pull Request magento/magento2#16247 from vgelani/magento2:typo-error-2
 - Merged commits:
   1. 3ca26ebc4e759ecb49a7faf48d57992b896c59a4

itaymesh committed:
Update Israeli ZIP code mask, 7 digits instead of 5 ,according to the Israeli postal office

svitja committed:
MAGETWO-92504: Products sorting in category is reverted to sort by product ID

svitja committed:
Merge remote-tracking branch 'origin/MAGETWO-92504' into 2.2-develop-pr31

OlgaVasyltsun committed:
MAGETWO-91163: Image doesn't load on frontend PDP if the product name contains double quote "

RuslanKostiv1 committed:
Merge remote-tracking branch 'mainline/2.2-develop' into MAGETWO-87589

RuslanKostiv1 committed:
Merge remote-tracking branch 'mainline/2.2-develop' into MAGETWO-75530

zakdma committed:
Merge remote-tracking branch 'origin/2.2-develop' into 2.2-develop-pr31

magento-engcom-team committed:
Merge pull request https://github.com/magento/magento2/pull/16241 to https://github.com/magento/magento2/pull/16240

magento-engcom-team committed:
Merge pull request https://github.com/magento/magento2/pull/16244 to https://github.com/magento/magento2/pull/16240

victor-v-rad committed:
Merge remote-tracking branch 'mainline/2.2-develop' into MAGETWO-74021

RuslanKostiv1 committed:
Merge remote-tracking branch 'mainline/2.2-develop' into MAGETWO-89238-2.2

Commit:
merge magento/2.2-develop into magento-tango/MAGETWO-74021

victor-v-rad committed:
Merge branch 'MAGETWO-74021' of https://github.com/magento-tango/magento2ce into MAGETWO-74021

magento-engcom-team committed:
ENGCOM-2057: Removed double occurrence of 'it' from sentences. #16240

 - Merge Pull Request magento/magento2#16240 from NamrataChangani/magento2:nps-double-occurrence-fix-1
 - Merged commits:
   1. eb3a6d25689e9de5b30503f19a6ce9783494ce69
   2. 139c0e14a500c596196f8be1a06012c294be11db
   3. 636587c3e0bce51edfe675fadda89bc843c792a1
   4. 2c28ddca8c8fffd62512a371265e79aa5071efc7
   5. 0fd2f9f4f1096f52786a21d6dd61bfd9b4777111
   6. 15ca0a2c73badbdfc92f25ccc6852e51282d5625

VladimirZaets committed:
Fix Magento/Framework/Code/Test/Unit/Generator/InterfaceGeneratorTest.php test

iivashchenko committed:
MAGETWO-92502: Extra XHR POST request made on every page

IvanPletnyov committed:
ENGCOM-1703: Move breadcrumb json configuration to viewmode (fix double quotes)

magento-engcom-team committed:
ENGCOM-2071: Fix missing PHPDocs hinting for AdvancedPricingImportExport module #15872

 - Merge Pull Request magento/magento2#15872 from dmytro-ch/magento2:fix/2.2-phpdocs-advanced-pricing-import-export
 - Merged commits:
   1. 3b9cda59b0900f9f621734e2b0417b6fa36674e2

iivashchenko committed:
Merge remote-tracking branch 'mainline/2.2-develop' into MAGETWO-92502

Commit:
MAGETWO-61209: Fixed issue with mage/calendar when setting `numberOfMonths` to show more than 1 month

(cherry picked from commit 54e1a715be3ef0bdfdb45b6fde99912d6e19fca6)

magento-engcom-team committed:
ENGCOM-2073: Update Israeli ZIP code mask: 7 digits instead of 5 #16250

 - Merge Pull Request magento/magento2#16250 from studioraz/magento2:impovment/update-israeli-zip-code-mask
 - Merged commits:
   1. 6345bbd5d625b9c4d73da59256d46fd9dfe41ba3

duhon committed:
Merge remote-tracking branch 'mainline/2.2-develop' into MAGETWO-92693

duhon committed:
MAGETWO-92693: Some improvements on product create|edit page in admin area

Commit:
Add UpdatedAtListProvider to NotSyncedDataProvider for invoice grid

Invoice grid isn't updated when state is changed.

magento-engcom-team committed:
ENGCOM-2078: Postpone instantiation of session config by using a proxy #15929

 - Merge Pull Request magento/magento2#15929 from fmarangi/magento2:15205-init-session-config
 - Merged commits:
   1. d23c198b4a7e9f5237d19e999cb160ac4059e61e
   2. b0b82d0025d0556e0a05951c54c997f45623015a
   3. 7e835c3cdecf70cf36c36968d4cd4bac7574b3eb

zakdma committed:
MAGETWO-87721: Custom Options are corruputed when saving product to a different website (fix create bundle product func test)

IvanPletnyov committed:
ENGCOM-2073: Update Israeli ZIP code mask: 7 digits instead of 5 #16250 (fix test)

OlgaVasyltsun committed:
MAGETWO-91163: Image doesn't load on frontend PDP if the product name contains double quote "

OlgaVasyltsun committed:
Merge remote-tracking branch 'origin/2.2-develop' into MAGETWO-91163

duhon committed:
MAGETWO-92693: Some improvements on product create|edit page in admin area

VladimirZaets committed:
ENGCOM-2057: Removed double occurrence of 'it' from sentences. #16240

VladimirZaets committed:
ENGCOM-2055: Fixed typo error #16247

VladimirZaets committed:
ENGCOM-2045: Fix case mismatch call (class/method) #16141

VladimirZaets committed:
ENGCOM-2042: Fix #15627: Product order in category changed in Magento 2.2.4 #15629

VladimirZaets committed:
ENGCOM-2034: PHPDoc #16215

VladimirZaets committed:
ENGCOM-1911: FIX for issue #15501 - M2.2.4 missing meta title tag and doesn't show… #15532

VladimirZaets committed:
ENGCOM-1892: Create ability to set is_visible_on_front to order status history comment #15637

VladimirZaets committed:
ENGCOM-1703: Move breadcrumb json configuration to viewmodel #15521

RuslanKostiv1 committed:
Merge remote-tracking branch 'origin/MAGETWO-75530' into PANDA-PR-2.2

RuslanKostiv1 committed:
Merge remote-tracking branch 'origin/MAGETWO-89238-2.2' into PANDA-PR-2.2

OlgaVasyltsun committed:
MAGETWO-91163: Image doesn't load on frontend PDP if the product name contains double quote "

rogyar committed:
Added integration test for checking admin login attempts cleanup

zakdma committed:
MAGETWO-87721: Custom Options are corruputed when saving product to a different website (fix endline in files)

magento-engcom-team committed:
ENGCOM-2085: Add UpdatedAtListProvider to NotSyncedDataProvider for invoice grid #16286

 - Merge Pull Request magento/magento2#16286 from JeroenVanLeusden/magento2:JeroenVanLeusden-patch-1
 - Merged commits:
   1. bceb4a34dd49dec602758d1f4517a59a7c74af53

zakdma committed:
MAGETWO-87721: Custom Options are corruputed when saving product to a different website (fix endline in files)

Commit:
merge magento/2.2-develop into magento-tsg/2.2-develop-pr31

Commit:
ENGCOM-2073: Update Israeli ZIP code mask: 7 digits instead of 5 #16250

rogyar committed:
Added integration test for checking customer login attempts captcha cleanup

magento-engcom-team committed:
[EngCom] Public Pull Requests - 2.2-develop

 - merged latest code from mainline branch

duhon committed:
Merge remote-tracking branch 'mainline/2.2-develop' into MAGETWO-92693

rogyar committed:
Added integration test for checking customer login attempts captcha cleanup after account edit

rogyar committed:
Minor improvements

slavvka committed:
DEVOPS-2632: Sample Data Builds Migrated from Bamboo

slavvka committed:
DEVOPS-2632: Sample Data Builds Migrated from Bamboo

magento-engcom-team committed:
ENGCOM-2090: Captcha: Added integration tests for checking customer login attempts cleanup #16306

 - Merge Pull Request magento/magento2#16306 from rogyar/magento2:2.2-captcha-reset-frontend-it
 - Merged commits:
   1. 690b39f11b65dac5fd8ae829e48e8224b9627a88
   2. 1cb9476de048885fd0c1d2fb1665f756a89b40ee

magento-engcom-team committed:
ENGCOM-2087: Captcha: Added integration test for checking admin login attempts cleanup #16300

 - Merge Pull Request magento/magento2#16300 from rogyar/magento2:2.2-captcha-reset-admin-it
 - Merged commits:
   1. 8982abeac9807bb4bfe069ee8d50b661dbcd57b1
   2. c2832f360300166c4f37812e29d8913c42aa9d91

Commit:
#16273: Fix bug in method getUrlInStore() of product model

# Method $product->getUrlInStore() returning extremely long URLs, could be a bug

(cherry picked from commit 7558ac0a478af8fc008228077f9e5380612d4138)

magento-engcom-team commented:
Hi @vasilii-b. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me {$VERSION} instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://github.com/magento/magento2/wiki/Magento-Contributor-Assistant) documentation

duhon committed:
Merge pull request #2735 from magento-performance/MAGETWO-92693

[MAGETWO-92693] Some improvements on product create|edit page in admin area

magento-engcom-team committed:
[EngCom] Public Pull Requests - 2.2-develop

 - merged latest code from mainline branch

magento-engcom-team committed:
ENGCOM-2097: Update webapi.xml to fix typo #15845

 - Merge Pull Request magento/magento2#15845 from mhaack/magento2:patch-1
 - Merged commits:
   1. 55b476e284e56ce5f23dadb3b12d78569950750a

magento-engcom-team committed:
ENGCOM-1611: Fixed set template syntax in block file #15339

 - Merge Pull Request magento/magento2#15339 from NamrataChangani/magento2:template-issue-1
 - Merged commits:
   1. efd58e34af13c0497652e10ae2151dc9e6e55802
   2. 86b98956b30950f9e00817e6cea12849de888a94
   3. c1ce795be9bfd0834e6e53a5edf303666595cce1
   4. 5118361850fcffe0d94a52e3f524ea4e920e9cfa

Commit:
:arrows_clockwise: [EngCom] Public Pull Requests - 2.2-develop

Accepted Public Pull Requests:
 - magento/magento2#16240: Removed double occurrence of 'it' from sentences. (by @NamrataChangani)
 - magento/magento2#16247: Fixed typo error (by @vgelani)
 - magento/magento2#16141: Fix case mismatch call (class/method) (by @lfluvisotto)
 - magento/magento2#15629: Fix #15627: Product order in category changed in Magento 2.2.4 (by @dverkade)
 - magento/magento2#16215: PHPDoc (by @lfluvisotto)
 - magento/magento2#15532: FIX for issue #15501 - M2.2.4 missing meta title tag and doesn't show (by @phoenix128)
 - magento/magento2#15637: Create ability to set is_visible_on_front to order status history comment (by @markoshust)
 - magento/magento2#15521: Move breadcrumb json configuration to viewmodel (by @diedburn)


Fixed GitHub Issues:
 - magento/magento2#15627: Product order in category changed after update to Magento 2.2.4 (reported by @dverkade) has been fixed in magento/magento2#15629 by @dverkade in 2.2-develop branch
   Related commits:
     1. 8981225996e92af97671c67727be35af42591e94

 - magento/magento2#15501: M2.2.4 missing meta title tag and doesn't show product name if meta title is empty (reported by @Naturegreen) has been fixed in magento/magento2#15532 by @phoenix128 in 2.2-develop branch
   Related commits:
     1. 0a38b845508577e38d79280288a4fc716f2d1108
     2. 85ac3be16708f9b7084481e522d29320e4a315e3
     3. 432cf007f79937437620f18ace00010b1a61a4ef
     4. 1413d65946c2e3e076a0605d2e5c739e0b339d83

AnshuMishra17 committed:
Merge branch '2.2-develop' of https://github.com/magento/magento2 into 2.2-develop

zakdma committed:
Merge remote-tracking branch 'origin/2.2-develop' into 2.2-develop-pr31

Commit:
merge magento/2.2-develop into magento-panda/PANDA-PR-2.2

Commit:
Merge branch '2.2-develop' of github.com:magento/magento2ce into MAGETWO-74021

le0n4eg committed:
Merge remote-tracking branch 'mainline/2.2-develop' into MAGETWO-87589

magento-engcom-team committed:
ENGCOM-2093: Admin controller product set save refactor #16217

 - Merge Pull Request magento/magento2#16217 from AnshuMishra17/magento2:2.2-develop
 - Merged commits:
   1. 2db91868f51ae6323095b053c78d656b74e41157
   2. 9d306722508b858aed7c6e43a6624e2970fa263a

slackerzz committed:
Add 'metatitle' to metadata test

Commit:
Merge pull request #2739 from magento-tsg/2.2-develop-pr31

[TSG] Backporting for 2.2 (pr31) (2.2.6)

magento-engcom-team committed:
ENGCOM-2106: Add metadata title in unit test #16333

 - Merge Pull Request magento/magento2#16333 from slackerzz/magento2:backport-11368
 - Merged commits:
   1. 5a0795f8799173bdbf8bc99bf1cc846ff98ec3b7

magento-engcom-team committed:
ENGCOM-2107: MAGETWO-61209: Backport - Fixed issue #7379 with mage/calendar when setting \`numberOfM… #16279

 - Merge Pull Request magento/magento2#16279 from vasilii-b/magento2:MAGETWO-61209-calendar-widget-number-of-months_mage22
 - Merged commits:
   1. de533cca86414906fc04302a7b7e3ec92186d85a

slavvka committed:
Merge remote-tracking branch 'mainline/2.2-develop' into DEVOPS-2632-2.2

slavvka committed:
Merge pull request #2747 from magento-borg/DEVOPS-2632-2.2

Fixed issues:
- DEVOPS-2632 Sample Data Builds Migrated from Bamboo

Commit:
ENGCOM-2097: Update webapi.xml to fix typo #15845

Commit:
ENGCOM-2090: Captcha: Added integration tests for checking customer login attempts cleanup #16306

Commit:
ENGCOM-2087: Captcha: Added integration test for checking admin login attempts cleanup #16300

Commit:
ENGCOM-2085: Add UpdatedAtListProvider to NotSyncedDataProvider for invoice grid #16286

Commit:
ENGCOM-2078: Postpone instantiation of session config by using a proxy #15929

Commit:
ENGCOM-2071: Fix missing PHPDocs hinting for AdvancedPricingImportExport module #15872

Commit:
ENGCOM-2007: [Backport] Fix issue #14895 - Change Password warning message appear two times #15774

Commit:
ENGCOM-1443: magento/magento2#12695: Unable to change attribute type from swatch #12771

Commit:
:arrows_clockwise: [EngCom] Public Pull Requests - 2.2-develop

Accepted Public Pull Requests:
 - magento/magento2#15845: Update webapi.xml to fix typo (by @mhaack)
 - magento/magento2#16306: Captcha: Added integration tests for checking customer login attempts cleanup (by @rogyar)
 - magento/magento2#16300: Captcha: Added integration test for checking admin login attempts cleanup (by @rogyar)
 - magento/magento2#16286: Add UpdatedAtListProvider to NotSyncedDataProvider for invoice grid (by @JeroenVanLeusden)
 - magento/magento2#15929: Postpone instantiation of session config by using a proxy (by @fmarangi)
 - magento/magento2#15872: Fix missing PHPDocs hinting for AdvancedPricingImportExport module (by @dmytro-ch)
 - magento/magento2#15774: [Backport] Fix issue #14895 - Change Password warning message appear two times (by @sanjay-wagento)
 - magento/magento2#12771: magento/magento2#12695: Unable to change attribute type from swatch (by @eugene-shab)


Fixed GitHub Issues:
 - magento/magento2#15205: Upgraded to Magento 2.2.4 from Magento 2.1.9 - Locale and Store Configuration issues 'Store View' Locale not being used on frontend, 'Default Config' Locale being used instead   (reported by @callumstar) has been fixed in magento/magento2#15929 by @fmarangi in 2.2-develop branch
   Related commits:
     1. d23c198b4a7e9f5237d19e999cb160ac4059e61e
     2. b0b82d0025d0556e0a05951c54c997f45623015a
     3. 7e835c3cdecf70cf36c36968d4cd4bac7574b3eb

 - magento/magento2#15245: 2.2.4: Wrong home page loaded in multi store setup  (reported by @jokeputs) has been fixed in magento/magento2#15929 by @fmarangi in 2.2-develop branch
   Related commits:
     1. d23c198b4a7e9f5237d19e999cb160ac4059e61e
     2. b0b82d0025d0556e0a05951c54c997f45623015a
     3. 7e835c3cdecf70cf36c36968d4cd4bac7574b3eb

 - magento/magento2#14895: Change Password warning message appear two times (reported by @cnviradiya) has been fixed in magento/magento2#15774 by @sanjay-wagento in 2.2-develop branch
   Related commits:
     1. b2cc966da3a6bcc925035e12a950e636847326f8
     2. cfe58e16b5159f97fbf8b1246a6e5d75caaaf9e0

 - magento/magento2#9307: Color attribute taking swatch instead of Drop down option for configurable options, (reported by @vishveskrish) has been fixed in magento/magento2#12771 by @eugene-shab in 2.2-develop branch
   Related commits:
     1. 39b1ef3a5b5a4741a0c664135b13944c4e7c7b61
     2. 94539b755992f120927a4760c06c624a0ced07c7
     3. 3d76096a752ff234843890d7a072cd1d64e8c1c2
     4. 0b079e05ab2e12c004fb1e81bda57817e5f0f14f
     5. 71d30b7c521fd5143a4cfd3252fa5c7c2c4fdd79
     6. 47b2f4841ae27fa7bde0124e8860493e79c62a92
     7. 707f1cd34f0f4c905b07658849984d0eb825d6d7
     8. 9153fe221d7ce9e65a3b082ab9d79c26245f1e0b
     9. 1bd4acfee8a58b035eef561aaebe94f97990a4a2
     10. f4990277bf9f20227402f4b76e57499873b170ad
     11. 3fa9ed2485a97972133ddb1f7d7e775f784fe6e4
     12. 8fa792a1444670e60fe6052c77e7b35684549906
     13. 7cd7adecf598dd6f9fe6ecd4369208869442387d
     14. 54ad76c1f48eafbfe16a4e8e3c7669145a190627
     15. bd126512cfe4f139a10ada9136495c1a617df158
     16. cb6975c3f57739946cb65acebaf62300def1fa99

 - magento/magento2#9923: Upgrading to 2.1.7 changed dropdown attributes to swatches (reported by @roseofgold) has been fixed in magento/magento2#12771 by @eugene-shab in 2.2-develop branch
   Related commits:
     1. 39b1ef3a5b5a4741a0c664135b13944c4e7c7b61
     2. 94539b755992f120927a4760c06c624a0ced07c7
     3. 3d76096a752ff234843890d7a072cd1d64e8c1c2
     4. 0b079e05ab2e12c004fb1e81bda57817e5f0f14f
     5. 71d30b7c521fd5143a4cfd3252fa5c7c2c4fdd79
     6. 47b2f4841ae27fa7bde0124e8860493e79c62a92
     7. 707f1cd34f0f4c905b07658849984d0eb825d6d7
     8. 9153fe221d7ce9e65a3b082ab9d79c26245f1e0b
     9. 1bd4acfee8a58b035eef561aaebe94f97990a4a2
     10. f4990277bf9f20227402f4b76e57499873b170ad
     11. 3fa9ed2485a97972133ddb1f7d7e775f784fe6e4
     12. 8fa792a1444670e60fe6052c77e7b35684549906
     13. 7cd7adecf598dd6f9fe6ecd4369208869442387d
     14. 54ad76c1f48eafbfe16a4e8e3c7669145a190627
     15. bd126512cfe4f139a10ada9136495c1a617df158
     16. cb6975c3f57739946cb65acebaf62300def1fa99

 - magento/magento2#11403: Product Attributes Not Updating on Frontend (reported by @rbur0425) has been fixed in magento/magento2#12771 by @eugene-shab in 2.2-develop branch
   Related commits:
     1. 39b1ef3a5b5a4741a0c664135b13944c4e7c7b61
     2. 94539b755992f120927a4760c06c624a0ced07c7
     3. 3d76096a752ff234843890d7a072cd1d64e8c1c2
     4. 0b079e05ab2e12c004fb1e81bda57817e5f0f14f
     5. 71d30b7c521fd5143a4cfd3252fa5c7c2c4fdd79
     6. 47b2f4841ae27fa7bde0124e8860493e79c62a92
     7. 707f1cd34f0f4c905b07658849984d0eb825d6d7
     8. 9153fe221d7ce9e65a3b082ab9d79c26245f1e0b
     9. 1bd4acfee8a58b035eef561aaebe94f97990a4a2
     10. f4990277bf9f20227402f4b76e57499873b170ad
     11. 3fa9ed2485a97972133ddb1f7d7e775f784fe6e4
     12. 8fa792a1444670e60fe6052c77e7b35684549906
     13. 7cd7adecf598dd6f9fe6ecd4369208869442387d
     14. 54ad76c1f48eafbfe16a4e8e3c7669145a190627
     15. bd126512cfe4f139a10ada9136495c1a617df158
     16. cb6975c3f57739946cb65acebaf62300def1fa99

 - magento/magento2#11703: Changing Swatches to Drop-down does not remove swatches from existing products (reported by @expgabe) has been fixed in magento/magento2#12771 by @eugene-shab in 2.2-develop branch
   Related commits:
     1. 39b1ef3a5b5a4741a0c664135b13944c4e7c7b61
     2. 94539b755992f120927a4760c06c624a0ced07c7
     3. 3d76096a752ff234843890d7a072cd1d64e8c1c2
     4. 0b079e05ab2e12c004fb1e81bda57817e5f0f14f
     5. 71d30b7c521fd5143a4cfd3252fa5c7c2c4fdd79
     6. 47b2f4841ae27fa7bde0124e8860493e79c62a92
     7. 707f1cd34f0f4c905b07658849984d0eb825d6d7
     8. 9153fe221d7ce9e65a3b082ab9d79c26245f1e0b
     9. 1bd4acfee8a58b035eef561aaebe94f97990a4a2
     10. f4990277bf9f20227402f4b76e57499873b170ad
     11. 3fa9ed2485a97972133ddb1f7d7e775f784fe6e4
     12. 8fa792a1444670e60fe6052c77e7b35684549906
     13. 7cd7adecf598dd6f9fe6ecd4369208869442387d
     14. 54ad76c1f48eafbfe16a4e8e3c7669145a190627
     15. bd126512cfe4f139a10ada9136495c1a617df158
     16. cb6975c3f57739946cb65acebaf62300def1fa99

 - magento/magento2#12695: Unable to change attribute type from swatch to dropdown  (reported by @alena-marchenko) has been fixed in magento/magento2#12771 by @eugene-shab in 2.2-develop branch
   Related commits:
     1. 39b1ef3a5b5a4741a0c664135b13944c4e7c7b61
     2. 94539b755992f120927a4760c06c624a0ced07c7
     3. 3d76096a752ff234843890d7a072cd1d64e8c1c2
     4. 0b079e05ab2e12c004fb1e81bda57817e5f0f14f
     5. 71d30b7c521fd5143a4cfd3252fa5c7c2c4fdd79
     6. 47b2f4841ae27fa7bde0124e8860493e79c62a92
     7. 707f1cd34f0f4c905b07658849984d0eb825d6d7
     8. 9153fe221d7ce9e65a3b082ab9d79c26245f1e0b
     9. 1bd4acfee8a58b035eef561aaebe94f97990a4a2
     10. f4990277bf9f20227402f4b76e57499873b170ad
     11. 3fa9ed2485a97972133ddb1f7d7e775f784fe6e4
     12. 8fa792a1444670e60fe6052c77e7b35684549906
     13. 7cd7adecf598dd6f9fe6ecd4369208869442387d
     14. 54ad76c1f48eafbfe16a4e8e3c7669145a190627
     15. bd126512cfe4f139a10ada9136495c1a617df158
     16. cb6975c3f57739946cb65acebaf62300def1fa99

Commit:
merge magento/2.2-develop into magento-tango/MAGETWO-74021

victor-v-rad committed:
Merge remote-tracking branch 'mainline/2.2-develop' into MAGETWO-74021

victor-v-rad committed:
Merge branch 'MAGETWO-74021' of https://github.com/magento-tango/magento2ce into MAGETWO-74021

Commit:
Merge branch 'MAGETWO-86658' into mpi-PR-2506

Commit:
Merge branch 'MAGETWO-92280' into mpi-PR-2506

Commit:
Merge branch 'MAGETWO-92502' into mpi-PR-2506

Commit:
Merge branch 'MAGETWO-92820' into mpi-PR-2506

magento-engcom-team committed:
ENGCOM-2122: When searching for the title if search for all the segments that has … #16093

 - Merge Pull Request magento/magento2#16093 from rsantellan/magento2:2.2-develop
 - Merged commits:
   1. 5c3ff9c485e7291cfe9e6c8923832206df8b42df

dhorytskyi committed:
Merge pull request #2725 from magento-tango/MAGETWO-74021

MAGETWO-74021: "Catalog Products List" widget does not displays on frontend

igrybkov committed:
Merge remote-tracking branch 'mainline/2.2-develop' into PANDA-PR-2.2

hitesh-wagento committed:
[Changed password placeholder text in checkout page]

le0n4eg committed:
MAGETWO-87589: [Magento Cloud] - Issue with polluted database when updating product attributes through API

Commit:
Merge branch '2.2-develop' of github.com:magento/magento2ce into mpi-PR-2506

IvanPletnyov committed:
ENGCOM-1611: Fixed set template syntax in block file #15339 (fix tests)

vrann committed:
Merge remote-tracking branch 'mainline/2.2-develop' into bulk-api-backport

vrann committed:
Merge remote-tracking branch 'bulk/bulk-api-backport' into bulk-api-backported

IvanPletnyov committed:
ENGCOM-1611: Fixed set template syntax in block file #15339 (fix tests)

le0n4eg committed:
MAGETWO-89238: [Sales email] Disabled invoice emails produce performance issue

le0n4eg committed:
Merge remote-tracking branch 'origin/MAGETWO-89238-2.2' into PANDA-PR-2.2

magento-engcom-team committed:
ENGCOM-2132: [Backport 2.2] Use correct error message for duplicate error key in product import #16389

 - Merge Pull Request magento/magento2#16389 from gelanivishal/magento2:2.2-develop-PR-port-15980
 - Merged commits:
   1. 9c9ac11f383fd9572c846186d6b658611bf4b054
   2. fa7801f72d127ebf65d05c2ccd50141e02e2c8c8

igrybkov committed:
Merge pull request #2752 from magento-mpi/mpi-PR-2506

Fixed issues:
  - MAGETWO-86658: Admin global search preview works inconsistently.
  - MAGETWO-92820: Once Payflow Pro payment is Declined, customer cannot continue checkout
  - MAGETWO-92280: [B2B] Unable to add to cart grouped product by SKU with enabled Shared Catalog
  - MAGETWO-92502: Extra XHR POST request made on every page

igrybkov committed:
Merge remote-tracking branch 'mainline/2.2-develop' into PANDA-PR-2.2

igrybkov committed:
Merge pull request #2743 from magento-panda/PANDA-PR-2.2

Fixed issues:
  - MAGETWO-89238: [Sales email] Disabled invoice emails produce performance issue
  - MAGETWO-75530: Deadlock when 'Asynchronous indexing' is enabled (insert from select on sales_order_grid).

orlangur committed:
Polish up implementation

le0n4eg committed:
Merge remote-tracking branch 'mainline/2.2-develop' into MAGETWO-87589

zakdma committed:
MAGETWO-85682: Default option for 'Status' attribute not being set

zakdma committed:
Merge remote-tracking branch 'origin/MAGETWO-91163' into 2.2-develop-pr32

zakdma committed:
Merge remote-tracking branch 'origin/MAGETWO-83992' into 2.2-develop-pr32

zakdma committed:
Merge remote-tracking branch 'origin/MAGETWO-92751' into 2.2-develop-pr32

zakdma committed:
Merge remote-tracking branch 'origin/MAGETWO-81926' into 2.2-develop-pr32

zakdma committed:
Merge remote-tracking branch 'origin/MAGETWO-85682' into 2.2-develop-pr32

magento-engcom-team committed:
ENGCOM-2143: Use constant time string comparison in FormKey validator #13509

 - Merge Pull Request magento/magento2#13509 from p0pr0ck5/magento2:form_key-constant-time
 - Merged commits:
   1. 506302fb5b474b9e312b732ee277218ccd89fe91
   2. 7d91cf4acb7e039f56a94a13d1af8c2021f30b29

vrann committed:
Merge remote-tracking branch 'mainline/2.2-develop' into bulk-api-backport

vrann committed:
Merge remote-tracking branch 'bulk/bulk-api-backport' into bulk-api-backported

svitja committed:
MAGETWO-81926: Category hierarchy isn't updated on Product Admin page

svitja committed:
Merge remote-tracking branch 'origin/MAGETWO-81926' into 2.2-develop-pr32

le0n4eg committed:
MAGETWO-87589: [Magento Cloud] - Issue with polluted database when updating product attributes through API

Commit:
Merge pull request #2768 from magento-tsg/2.2-develop-pr32

[TSG] Backporting for 2.2 (pr32) (2.2.6)

Commit:
merge magento/2.2-develop into magento-engcom/bulk-api-backported

hitesh-wagento committed:
[Added Password text in .csv file]

vrann committed:
Merge remote-tracking branch 'mainline/2.2-develop' into bulk-api-backported

vrann committed:
Merge branch 'bulk-api-backported' of https://github.com/magento-engcom/magento2ce into bulk-api-backported

le0n4eg committed:
MAGETWO-87589: [Magento Cloud] - Issue with polluted database when updating product attributes through API

le0n4eg committed:
Merge remote-tracking branch 'mainline/2.2-develop' into MAGETWO-87589

vrann committed:
Merge pull request #2767 from magento-engcom/bulk-api-backported

Bulk api backported

magento-engcom-team committed:
ENGCOM-2126: [Changed password placeholder text in checkout page] #16379

 - Merge Pull Request magento/magento2#16379 from hitesh-wagento/magento2:wrong-password-checkout
 - Merged commits:
   1. 4f976e703be9390cf4080050d309dd0510ab11a4
   2. 55d1c1573eaa63c4d307e498289da0c812e6ab27

Commit:
ENGCOM-2143: Use constant time string comparison in FormKey validator #13509

Commit:
ENGCOM-2132: [Backport 2.2] Use correct error message for duplicate error key in product import #16389

Commit:
ENGCOM-2126: [Changed password placeholder text in checkout page] #16379

Commit:
ENGCOM-2122: When searching for the title if search for all the segments that has … #16093

Commit:
ENGCOM-2107: MAGETWO-61209: Backport - Fixed issue #7379 with mage/calendar when setting `numberOfM… #16279

Commit:
ENGCOM-2106: Add metadata title in unit test #16333

Commit:
ENGCOM-2093: Admin controller product set save refactor #16217

Commit:
ENGCOM-1611: Fixed set template syntax in block file #15339

le0n4eg committed:
Merge remote-tracking branch 'mainline/2.2-develop' into MAGETWO-87589

magento-engcom-team committed:
:arrows_clockwise: [EngCom] Public Pull Requests - 2.2-develop

Accepted Public Pull Requests:
 - magento/magento2#13509: Use constant time string comparison in FormKey validator (by @p0pr0ck5)
 - magento/magento2#16389: [Backport 2.2] Use correct error message for duplicate error key in product import (by @gelanivishal)
 - magento/magento2#16379: [Changed password placeholder text in checkout page] (by @hitesh-wagento)
 - magento/magento2#16093: When searching for the title if search for all the segments that has  (by @rsantellan)
 - magento/magento2#16279: MAGETWO-61209: Backport - Fixed issue #7379 with mage/calendar when setting `numberOfM (by @vasilii-b)
 - magento/magento2#16333: Add metadata title in unit test (by @slackerzz)
 - magento/magento2#16217: Admin controller product set save refactor (by @AnshuMishra17)
 - magento/magento2#15339: Fixed set template syntax in block file (by @NamrataChangani)


Fixed GitHub Issues:
 - magento/magento2#16378: Wrong placeholder for password field in the checkout page (reported by @mohitka) has been fixed in magento/magento2#16379 by @hitesh-wagento in 2.2-develop branch
   Related commits:
     1. 4f976e703be9390cf4080050d309dd0510ab11a4
     2. 55d1c1573eaa63c4d307e498289da0c812e6ab27

 - magento/magento2#7379: Calendar widget (jQuery UI DatePicker) with numberOfMonths = 2 or more (reported by @atihomirov) has been fixed in magento/magento2#16279 by @vasilii-b in 2.2-develop branch
   Related commits:
     1. de533cca86414906fc04302a7b7e3ec92186d85a

 - magento/magento2#15345: Template syntax in block file (reported by @vgelani) has been fixed in magento/magento2#15339 by @NamrataChangani in 2.2-develop branch
   Related commits:
     1. efd58e34af13c0497652e10ae2151dc9e6e55802
     2. 86b98956b30950f9e00817e6cea12849de888a94
     3. c1ce795be9bfd0834e6e53a5edf303666595cce1
     4. 5118361850fcffe0d94a52e3f524ea4e920e9cfa

le0n4eg committed:
Merge remote-tracking branch 'mainline/2.2-develop' into MAGETWO-87589

arhiopterecs committed:
Merge pull request #2774 from magento-panda/MAGETWO-87589

Fixed issues:
 - MAGETWO-87589: [Magento Cloud] - Issue with polluted database when updating product attributes through API

Commit:
Modified RouteParamsResolverTest according to latest bugfixes on tested class

Commit:
Merge remote-tracking branch 'upstream/#16273-product-geturlinstore-wrong-link_mage22' into #16273-product-geturlinstore-wrong-link_mage22

magento-cicd2 commented:
[![CLA assistant check](https://cla.dev.magento.com/pull/badge/not_signed)](https://cla.dev.magento.com/magento/magento2?pullRequest=16308) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/magento2?pullRequest=16308) before we can accept your contribution.<br/>**2** out of **8** committers have signed the CLA.<br/><br/>:white_check_mark: fmarangi<br/>:white_check_mark: magento-engcom-team<br/>:x: victor-v-rad<br/>:x: le0n4eg<br/>:x: RuslanKostiv1<br/>:x: svitja<br/>:x: DianaRusin<br/>:x: zakdma
